### ISSUE TITLE
[no-unused-vars] get type of a interface property

### ISSUE THREAD

**Issue by wenkanglin** [bradzacher/eslint-plugin-typescript#256]
**Repro**

```JSON
{
  "rules": {
    "no-unused-vars": "error",
    "typescript/no-unused-vars": "error"
  }
}
```

```TS
import { PermissionItem } from '@/types/permission'

let a: PermissionItem['name'] = 'test'

console.log(a)
```

**Expected Result**

The `PermissionItem` is defined and used

**Actual Result**

[eslint] 'PermissionItem' is defined but never used. (no-unused-vars)

**Additional Info**

When I import a interface and get type of a property from it used as type of a variable, it will catch eslint `no-unused-vars` error

**Versions**

| package                    | version |
| -------------------------- | ------- |
| `eslint-plugin-typescript` | `0.14.0` |
| `typescript-eslint-parser` | `21.0.2` |
| `typescript`               | `3.2.2` |

**Comment by armano2** [comment]
this issue got fixed in upcoming [v1 version ](https://github.com/bradzacher/eslint-plugin-typescript/milestone/1)
